tools/cli/NearObjectCli.cxx

```diff
@@ -483,6 +545,40 @@ NearObjectCli::AddSubcommandUwbRangeStart(CLI::App* parent)
             m_cliData->appConfigParamsData.destinationMacAddresses.value().insert(uwb::UwbMacAddress::FromString(m_cliData->destinationMacAddressString, macAddressType).value());
         }
 
+        // Parse and validate ResultReportConfig
+        constexpr int resultReportConfigurationSize = 4; // TODO: Find a way to calculate this number
+        auto IsValidResultReportConfigurationString = [resultReportConfigurationSize](const std::string& resultReportConfigurationString) {
+            if (resultReportConfigurationString.length() != resultReportConfigurationSize) {
+                std::cerr << "Invalid ResultReportConfiguration length" << std::endl;
+                return false;
+            }
+            for (char c : resultReportConfigurationString) {
+                if (c != '0' && c != '1') {
+                    std::cerr << "Invalid ResultReportConfiguration value" << std::endl;
+                    return false;
+                }
+            }
+            return true;
+        };
+
+        if (IsValidResultReportConfigurationString(m_cliData->resultReportConfigurationString)) {
+            m_cliData->appConfigParamsData.resultReportConfig.emplace();
+
+            const std::bitset<4> resultReportConfigurationBits(m_cliData->resultReportConfigurationString);
+            if (resultReportConfigurationBits[0]) {
```

Comment:
FYI, `std::bitset` also has a `test()` function is does the same thing as the subscript operator here. I like that `test()` more directly indicates what's being done, but I'm mostly bringing this up for informational purposes. Feel free to close.